src/main/kotlin/org/rust/ide/intentions/ContractModule.kt

```diff
@@ -0,0 +1,32 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.intellij.psi.impl.file.PsiFileImplUtil
+import com.intellij.refactoring.move.moveFilesOrDirectories.MoveFilesOrDirectoriesUtil
+import org.rust.lang.core.psi.impl.RustFileImpl
+import org.rust.lang.core.psi.util.RustModules
+
+class ContractModule : IntentionAction {
+
+    override fun getFamilyName()        = "Contract module structure"
+    override fun getText()              = "Contract Module"
+    override fun startInWriteAction()   = true
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?) =
+        file is RustFileImpl
+            && file.name == RustModules.MOD_RS
+            && file.containingDirectory?.children?.size == 1
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?) {
+        file ?: return
+        val dst = file.parent?.parent ?: return
+        val parent = file.parent ?: return
+        val fileName = parent.name + ".rs"
```

Comment:
A couple of more minor stylistic changes:

* use string interpolation for `fileName`,
* move `val parent` before `val dst` and use `parent to calculate `dst`.